### DISCUSSION TITLE
(Solved) Easier way to haul empty fuel rods, empty ammunition boxes and empty welding fuel tanks out of the ship, or a refill option on stations (just like maintenance), or an incinerator on the sub

### DISCUSSION THREAD

**Discussion by RodrigoGuides** [FakeFishGames/Barotrauma#4830] [Ideas and suggestions]
**EDIT:** Well this has been addressed with the new, upcoming Cave Update, since now items will be stacked. Since the issue of my topic has been addressed, I will consider this topic to be finished. (I can't delete it, but I won't respond anymore because the issue has been addressed already)

**ORIGINAL TOPIC:**

Every few missions, a lot of empty items accumulate in the sub (empty fuel rods, empty ammunition boxes and empty welding fuel tanks). If you play only with bots or if other people won't help you haul these items, they can accumulate pretty fast.

I can personally haul these empty items to discard them on stations, or use a Deconstructor to salvage them and craft new items, but these both proccesses take a lot of time of travelling and manual hauling. I only play singleplayer with bots, so there are no other players to help me haul these items, and the bots won't do it just yet. And the empty items **have** to be hauled out of the ship, because otherwise they will keep taking storage space (instead of new, filled items) and also cluttering the ship, which in an emergency could be difficult. It also makes it more difficult to have a good notion of how much fuel I have for the reactor and other calculations necessary for the missions.

**I proppose some solutions to this issue:**

1) Refill button on stations (working in the same way that the maintenance button which fixes all the hulls for 500 marks), this button would refill all empty items on the submarine (fuel rods, ammunition boxes and welding fuel tanks)
2) Bots automatically (or after being given the order to Clean) haul items to stations either to discard them or to deconstruct them, but they would also need to be able to craft and bring the new items back (and I'm sure this would take a ton of scripting, thus I think the button would be just easier to implement, or the next option)
3) Being able to sell empty items to traders on stations (and without the need to manually haul these items to the stations to be able to sell them). I don't mind if the selling price would be 1mk or 0mk, it's just to get rid of the empty items without having to manually haul them everytime.
4) Some kind of "refreshable" trash bin on the sub, where you can put empty items infinitely (this would still require hauling but it would mean less travelling; maybe bots could be instructed to take empty items to the trash). However, bots would need to be able to not take items from the trash and put it back in storage. If an infinite trash bin does not make sense lore-wise, maybe an incinerator instead, which would serve the same purpose, but all vanilla subs would need to start with one for QoL purposes.

**Comment by Kulik242** [comment]
Not sure if that's what Devs intented, but the whole "get the ship in order and run errands around outpost" is a part of the game. 
In our campaign, after each session, the sub is in complete disarray and I spent as much time sorting things, selling things, buying things and reorganising things as in mission. I don't mind but I can see how this may be a problem for others.

**Reply by RadCatish** [reply]
I agree, Barotrauma is a submarine management game, I feel like this is an important part of it

**Reply by RodrigoGuides** [reply]
Well, each one has their tastes, and I'm not trying to take away some people's choice to manually replace the fuel rods if they enjoy it, however, as you said yourself, you can see how this may be a problem for other people. Just to be clear, I'm referring to the rods stored in the ship, not the rods being used by the reactor.

When the hull's walls get damaged (by creatures or collision), during the voyage we have to repair them by ourselves. When we reach a station, we also have the option to pay around 500 in money to get them fully repaired. Having more options is always good, and no one has to use them if they don't want to.

Having the option to have the walls fully repaired doesn't take away from the joy of repairing the hulls mid-voyage, and it is useful if for example your crew has forgotten a small part of it or is unaware of some minor damage.

I know some people could get "offended" by this suggestion, but no one would have to use such option if they don't want to, it would just make the game more approachable to the people who don't want to be manually hauling empty fuel rods and new fuel rods between the ship and the stations.

**Reply by mLuby** [reply]
FWIW I don't think I've _ever_ used those "Repair all" or "Fix all" buttons. I assumed they're there in case the sub has broken yet inaccessible walls or machines.

However, as you say, the presence of those buttons doesn't hurt my gameplay experience; it's just a feature I never use that burned some development time.

**Reply by RodrigoGuides** [reply]
I wouldn't say it burned development time, since for some people it could be important and make a difference in deciding to keep playing or not (and thus avoiding a refund). Some people like micromanagement, and some don't. It's down to personal tastes.

I rarely use those Repair All buttons, but I recognize having them is much better than having to personally check every single corner of the ship everytime I have a doubt. Nevertheless, for the people who want to do it manually, they can.

As you said, you've never even used those buttons, so having the buttons isn't harmful to the playerbase that doesn't want to use them. It's just more options.

**Reply by mLuby** [reply]
"Burned" may have been too strong. I'm just saying that since the devs have limited bandwidth, they have to choose between alternatives that accomplish the same goal. So it's smart to implement things that most players will use, like the Bot `Repair Broken Devices` Order, _before_ things that most players won't use, like the `Fix All` button. Ideally there'd be time to do both.

So though it does no _harm_, adding a `Refuel Rods` button has a _cost_, and that's why I'm making a friendly 😇 argument against it in favor of alternatives.

**Comment by Sviat** [comment]
I would actually vote for recharge Installations, similar to Oxygen generator, that have input stash of (for example) Ethanol and inventory slots for 5 tanks, and they recharge welding fuel tanks.
Same for Fuel Rods.

Reason is - having Fabricator + Deconstructor means you can reproduce the very same alchemy sequence (Al + Eth = Fuel ; Empty Fuel -> Al; Al + Eth ... ) yourself, just wasting clicks.
Why if this is 3rd most recurring task?

Coilgun Ammunition boxes are harder, and usually, rotated less often, so no charging station for them.

**Reply by mLuby** [reply]
I like recharging stations for Welding Tanks and Ammo Boxes, but not for nuclear fuel rods (even setting aside how deathly radioactive they become after used).

That said, I'd still expect to pay for the welding fuel or bullets, so is it really much of an improvement over buying brand new ones?

**Comment by NilanthAnimosus** [comment]
Given stores only have certain amounts of uranium or fuel rods, materials and other such, having them recharge such items would end up netting you more than they may even have available to sell you usually. as well as trying to balance proper costs etc for the service or even to choose what you do and do not want recharged.

I do not think ammunition nor fuel/welding materials and such should be refillable due to the above point, at most only oxygen which the player can already do anyways. but I do very much recognize the tedium this can have if doing it alone especially clearing a submarine of used items.

I think welding fuel should have a method of combining tanks and refilling like oxygen tanks, perhaps by having the equivalent of a recharging dock for them, this would alleviate that and is a small QOL change. if it cannot be done with current item components I am sure a new item component class could be created just to handle item recharging with a consumable item (Such as ethanol) using two separate inventories to make that easily possible.

Fuel rods I think would be nice to combine as well, instead of directly deconstructing to get small % parts uranium out.

I saw a mod for combining coilgun ammunition which mostly worked fine. I don't know why this is not a default feature even if a bit odd to combine in another container such as a loader. the only issue I encountered with this was the coilgun box temporarily disappeared from being held - until you dropped it and picked it back up.

Now when it comes to box-goods like coilgun ammunition -  Perhaps if we could somehow bulk-sell empty coilgun ammo boxes straight from the submarine? that would be nice, carrying it to a merchant is indeed a time consuming pain if it was made sellable.

**Reply by RodrigoGuides** [reply]
Well, yes, if you could somehow sell empty boxes straight from the sub, even if for a small amount compared to the price of a new, filled box, it would already be a great QoL change, because when playing singleplayer (which I do), you have to haul everything yourself, as you said, is quite tedious. Being able to sell these empty items straight from the sub or at least not having to haul them directly, would already be a very welcome change.

However, I stand to the point that this should also apply to empty fuel rods and empty welding fuel tanks. All of these items have to be personally hauled at stations when playing singleplayer, and the bigger the ship, the longer it takes. I still suggest that there should be an alternative. While crafting and deconstructing is coherent lore-wise, it still requires personally hauling everything back and forth between the sub and the station. The hauling part is something the bots could do, or be skipped entirely.

**Reply by NilanthAnimosus** [reply]
Yeah, I've been playing with myself and 2 others, but they both do not sell nor craft or help me with the logistics side - I do agree that it should apply to other useless empty items, but they actually don't have sell prices when I carried spent uranium rods to the station (Much to my sadness) - If items had a scaled price range to durability for the item they need to refill it, or just some cheap price and a fast way to remove it, it would be good.

Though I don't think anyone wants the entire submarines inventory in every single container and sub container to appear in the store, maybe just some of the low durability items - perhaps even convert the item to a new "Depleted" variant or something or some identifier to make it sellable from anywhere if empty (Since its now useless and essentially just a chore to handle).

**Reply by RodrigoGuides** [reply]
Yeah, agreed with what you said.

I'm not trying to make the game easier, I just want to save us some time.

**Reply by mLuby** [reply]
> I think welding fuel should have a method of combining tanks and refilling like oxygen tanks, perhaps by having the equivalent of a recharging dock for them, this would alleviate that and is a small QOL change.

@NilanthAnimosus There is a mod that does what you want: [Welding Fuel Generator](https://steamcommunity.com/sharedfiles/filedetails/?id=2255427810). Enjoy!

> if you could somehow sell empty boxes straight from the sub, even if for a small amount compared to the price of a new, filled box, it would already be a great QoL change

If you deconstruct the boxes, you can carry the resulting steel with you and sell it for a small amount.

It sounds to me like you're both playing alone on subs without (de)constructors—so using station (de)constructor facilities is a pain due to travel distance. What if those machines were right by the station airlock entrance so you don't have to search for them or travel so far?

**Reply by RodrigoGuides** [reply]
Yes, the Deconstructor machines standing at the entrance would be helpful, but I wouldn't consider it a definitive solution since if my ship is big enough, I will still have to travel a lot to haul the items between the sub and the station. That is, if the station does have a Deconstructor.

As you suggested in another post, maybe it bots could simply carry the empty items to the station (and thus do the hauling by themselves), or if we could sell empty items to the station NPC instead of having to haul them, it would be great.

**Reply by Sviat** [reply]
> Given stores only have certain amounts of uranium or fuel rods, materials and other such, having them recharge such items would end up netting you more than they may even have available to sell you usually. as well as trying to balance proper costs etc for the service or even to choose what you do and do not want recharged.
> 
> I do not think ammunition nor fuel/welding materials and such should be refillable due to the above point, at most only oxygen which the player can already do anyways. but I do very much recognize the tedium this can have if doing it alone especially clearing a 
submarine of used items.

Well obviously not for free, and not by violation of Conservation Principles.
For example - read my post above. You below suggest similar approach. Having Fabricator on your Sub (who's sailing without one?) means you already can REFILL these items, with some consumables, like Ethanol and Uranium.
If you do not want machinery for refill, also fine for me, can be just another shortcut recipe for Fabricator - (Fuel Tank + Ethanol = Fuel Tank 100% Condition); plus similar for other two (Ammobox and Rod)

**Comment by mLuby** [comment]
> I only play singleplayer with bots, so I have to do all of the micromanaging of the sub's empty rods, boxes and tanks. I do not want to make the game easier, I just want to save time.

Personally I think **improving the Bot "Clean" Order would help the most here**, so that bots with that Order would deconstruct depleted fuel rods, welding tanks, and ammo boxes. That'd be great for both single _and_ multiplayer use cases, and wouldn't need changes to items or machines.

Therefore, while I appreciate the thought that's gone into writing your suggestion, 😇 I respectfully disagree with the autofill button as currently proposed.

**Reply by RodrigoGuides** [reply]
Well, yes, maybe the autofill button isn't the ideal, it's just easier to implement from a dev perspective, because a better cleaning function requires further scripting and I don't know where that is in the roadmap. A button could maybe be implemented in a smaller patch, while a more complex Clean order could take many months from now. Since I only play in singleplayer, this is a priority for me (thus I suggested a button which is easier to implement), but that's a personal standpoint.

Particularly I never suggested new machines or alterations to existing ones, or alterations to the items. I suggested an easier way to get rid of empty items and refill them. Other people have suggested changes to machines or items, as alternatives to the refill button I suggested.

I understand what you said, and I agree that with a more complex Clean order, the same solution would be achieved. Currently, unfortunately bots do not differentiate between empty items and filled items, and as the captain of a singleplayer campaign I constantly need to haul empty items out of the ship manually, so that filled items can take their place in the storages.

I changed the title to better reflect my proposal.

**Comment by RadCatish** [comment]
What about some sort of "dumbwaiter" crate shelf which, when you dock at a station, transfers all crates within it to the feet of the shop NPC. In the same way that cargo you buy auto-spawns in your cargo bay when you start a mission, items you've put in this location can auto-despawn to the station. Prevents going up and downstairs carrying multiple crates through the outpost, but if you've accidentally put something in there you didn't want to sell you can still choose not to when you arrive at the shop.

In our campaigns we always have a "For Sale" crate rack, and (lugging around outposts aside) it seems to work pretty well

**Reply by Sviat** [reply]
"For Sale or Dispose I Just Do Not Need These" box

**Reply by NilanthAnimosus** [reply]
That is an interesting concept, though somebody still has to haul them to said shelf. I don't think my AI's or friends will do that, so it cuts back the distance and only if your submarine just happens to contain this particular rack, many of which may not even have the space such as small submarines but still perhaps its fair share of garbage. still a nice idea potentially.

**Reply by Sviat** [reply]
> 
> That is an interesting concept, though somebody still has to haul them to said shelf. I don't think my AI's or friends will do that, so it cuts back the distance and only if your submarine just happens to contain this particular rack, many of which may not even have the space such as small submarines but still perhaps its fair share of garbage. still a nice idea potentially.

Should be doable with minimum intervention by utilizing "Clean" command and "Containable Restrictions" (currently no way to separate empty and full, but we just drop on the floor items we want bots to take care of)
![image](https://user-images.githubusercontent.com/2896936/105386258-7cede980-5c14-11eb-8153-48010da3464c.png)

**Reply by RodrigoGuides** [reply]
> What about some sort of "dumbwaiter" crate shelf which, when you dock at a station, transfers all crates within it to the feet of the shop NPC. In the same way that cargo you buy auto-spawns in your cargo bay when you start a mission, items you've put in this location can auto-despawn to the station. Prevents going up and downstairs carrying multiple crates through the outpost, but if you've accidentally put something in there you didn't want to sell you can still choose not to when you arrive at the shop.
> 
> In our campaigns we always have a "For Sale" crate rack, and (lugging around outposts aside) it seems to work pretty well

Yes, I like this idea.

My only intention really is to skip the time of manually hauling things in and out, and this idea would help a lot in that regard. I support this. Just maybe, don't drop it at the feet of the trader, since this would hinder clicking on the trader, maybe drop in a container or something on the station, I don't know, but this is a minor detail. I support the idea as a whole.

> > That is an interesting concept, though somebody still has to haul them to said shelf. I don't think my AI's or friends will do that, so it cuts back the distance and only if your submarine just happens to contain this particular rack, many of which may not even have the space such as small submarines but still perhaps its fair share of garbage. still a nice idea potentially.
> 
> Should be doable with minimum intervention by utilizing "Clean" command and "Containable Restrictions" (currently no way to separate empty and full, but we just drop on the floor items we want bots to take care of)

I do not know how to use "Containable Restrictions" or that stuff, but if we can get bots to put empty things on that disposable crate that RadFish suggested, it would be a huge step. But this is tricky since some people just want to get rid of empty things (like me) and other people want to save them for salvaging. Maybe people could either "activate" or "deactivate" such crate according to each playstyle?

**Comment by RodrigoGuides** [comment]
Well this has been addressed with the new, upcoming Cave Update, since now items will be stacked. Since the issue of my topic has been addressed, I will consider this topic to be finished. (I can't delete it, but I won't respond anymore because the issue has been addressed already)
